src/main/java/duke/taskList/TaskList.java

```diff
@@ -0,0 +1,358 @@
+            System.out.print("_");
+        }
+        System.out.println("");
+    }
+
+    /**
+     * Deletes a duke.task from ArrayList.
+     * @param command The duke.command specifying which duke.task to delete.
+     */
+    public String deleteTask(String command) {
+        String[] splitCommand = command.split(" ");
+        String strToPrint = "";
+        try {
+            if (splitCommand.length != 2) {
+                throw new DukeException("Please insert a numerical value to delete");
+            }
+            Pattern pattern = Pattern.compile("-?\\d+(\\.\\d+)?");
+            if (!pattern.matcher(splitCommand[1]).matches()) {
+                throw new DukeException("Please enter a numerical value");
+            } else {
+                if (Integer.parseInt(splitCommand[1]) > list.size()) {
+                    throw new DukeException("The delete value is out of range");
+                }
+            }
+        } catch (DukeException e) {
+            printHorizontalLine();
+            System.out.println(e.getMessage());
+            printHorizontalLine();
+            return e.getMessage();
+        }
+        strToPrint += ("Noted. I've removed this duke.task:\n");
+        int indexToRemove = Integer.parseInt(command.substring(7)) - 1;
+        Task taskToRemove = list.get(indexToRemove);
+        strToPrint += (taskToRemove) + "\n";
+        list.remove(indexToRemove);
+        try {
+            File myObj = new File("./src/main/data/tmpDuke.txt");
+            myObj.createNewFile();
+            int currLine = 0;
+            File currFile = new File("./src/main/data/duke.txt");
+            Scanner myReader = new Scanner(currFile);
+            while (myReader.hasNextLine()) {
+                if (indexToRemove == currLine) {
+                    String data = myReader.nextLine();
+                    currLine++;
+                    continue;
+                }
+                currLine++;
+                String data = myReader.nextLine();
+                if (data.equals("")) {
+                    break;
+                }
+                FileWriter myWriter = new FileWriter("./src/main/data/tmpDuke.txt", true);
+                myWriter.write(data + "\n");
+                myWriter.close();
+            }
+            myReader.close();
+            new File("./src/main/data/tmpDuke.txt").renameTo(currFile);
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+            return "An error occurred";
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+            return "An error occurred";
+        }
+        return strToPrint += ("Now you have " + list.size() + " tasks in the list.");
+    }
+
+
+    /**
+     * Adds a event into ArrayList.
+     * @param command The duke.command specifying what Event to add.
+     */
```

Comment:
Again, a @return tag here might be useful?

Other instances of @return tag can be added to other functions in this file as well!